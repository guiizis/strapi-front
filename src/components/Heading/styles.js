import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
  `,
};

const titleCase = (upperCase) => css`
  text-transform: ${upperCase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, upperCase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(upperCase)}
  `}
`;
