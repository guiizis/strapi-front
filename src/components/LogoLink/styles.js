import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: flex;
    align-itens: center;
    text-deoration: none;
    color: inherit;

    > img {
      max-height: 3rem;
    }
  `}
`;
