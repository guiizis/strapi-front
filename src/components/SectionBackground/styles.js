import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background && containerBackgroundActive(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;

const containerBackgroundActive = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;
