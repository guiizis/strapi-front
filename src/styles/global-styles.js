import { createGlobalStyle, css } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=Montserrat:ital@1&family=Open+Sans:ital,wght@0,400;0,700;1,400;1,600&display=swap');

* {
    margin: 0;
    padding: 0;
    box-sizing: 'border-box'
  }

  html {
    font-size: 62.5%;
  }

  body {
    font-size: 2.4rem;
    font-family: 'Open Sans', sans-serif;
  }

  h1 {
    font-family: 'Montserrat', sans-serif;
  }
`;
