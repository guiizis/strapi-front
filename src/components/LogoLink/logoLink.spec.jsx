import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { LogoLink } from '.';

describe('<LogoLink />', () => {
  it('should render', () => {
    renderTheme(<LogoLink link="#target" text="Olá Mundo" />);
    expect(
      screen.getByRole('heading', { name: 'Olá Mundo' }),
    ).toBeInTheDocument();
  });

  it('should render an image', () => {
    renderTheme(
      <LogoLink
        link="#target"
        text="Olá Mundo"
        image="assets/images/logo.svg"
      />,
    );

    const heading = screen.getByRole('heading', { name: 'Olá Mundo' });
    expect(heading.firstChild.firstChild).toHaveAttribute(
      'src',
      'assets/images/logo.svg',
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <LogoLink
        link="#target"
        text="Olá Mundo"
        image="assets/images/logo.svg"
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
