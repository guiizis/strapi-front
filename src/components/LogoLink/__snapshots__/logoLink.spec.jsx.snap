// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should match snapshot 1`] = `
.c0 {
  color: #0a1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-itens: center;
  -webkit-box-align-itens: center;
  -ms-flex-align-itens: center;
  align-itens: center;
  text-deoration: none;
  color: inherit;
}

.c1 > img {
  max-height: 3rem;
}

<h1
  class="c0"
>
  <a
    class="c1"
    href="#target"
  >
    <img
      alt="OlÃ¡ Mundo"
      src="assets/images/logo.svg"
    />
  </a>
</h1>
`;
